scripts/result_scanner/result_scanner.rb

```diff
@@ -0,0 +1,118 @@
+#!/usr/bin/env ruby
+require 'json'
+
+module ResultScanner
+
+  Result = Struct.new(:name, :confidence, :risk, :url, :param, :evidence, :alert)
+
+  class << self
+
+    # Call Me.
+    def summarize_results(last_run_path, current_run_path, output_path)
+      @final_json_out = {}
+      @final_text_out = "Completed scan of #{json_files_from_dir(current_run_path).size} properties:"
+
+      all_project_names(last_run_path, current_run_path).each do |proj|
+        last_json = read_json(file_path_from_project_name(proj, json_files_from_dir(last_run_path)))
+        current_json = read_json(file_path_from_project_name(proj, json_files_from_dir(current_run_path)))
+
+        last_results = create_results(proj, last_json)
+        current_results = create_results(proj, current_json)
+
+        deltas = compute_risk_level_deltas(last_results, current_results)
+        project_status = risk_level_delta_status_messages(deltas, last_json.empty?, current_json.empty?)
+
+        @final_json_out.merge!({ proj => (count_risk_levels(current_results).merge(status: project_status)) })
+        @final_text_out << "\n#{proj}: #{paren_status_count(current_results)} #{project_status}"
+      end
```

Comment:
Can we simplify this loop?

* Move things into smaller methods
* Use more variables rather than nesting function calls